### DISCUSSION TITLE
Copying images to clipboard

### DISCUSSION THREAD

**Discussion by casperl** [zim-desktop-wiki/zim-desktop-wiki#1691] [Ideas]
**Problem:**

I cannot find an easy way of copying an image to the clipboard where the image is included on a Notebook page or from the image icon in the attachment folder pane.

**Desired Functionality:**

The functionality I desire is to right-click on an image and select "copy image" that will place the image on the system clipboard to be pasted into a different application.

It will be a bonus if images can be copied by right-clicking on the image icon in the attachments folder and selecting "copy image".

**Results of trying existing functions:**

I tried selecting the image area in the wiki page body, pressing Ctrl-C and then pasting in the target application but that simply pastes a reference to the image as in 

./pasted_image.png

**Alternate method that works:**

I can go to the target application and attach Zim wiki images from the relevant folder but that soon becomes tedious.

There is also a "copy as" as option when right-clicking on an image, but the options under that are greyed out in the sub-selection box.

**Host Platform:**

I am using Zim 0.74.2 on Ubuntu 20.04

**Comment by Inconel625** [comment]
Would love to see this functionality added.

**Comment by hsandt** [comment]
> I can go to the target application and attach Zim wiki images from the relevant folder but that soon becomes tedious.

Indeed, and besides right-click > Open Folder only opens the folder without focusing the item so if you have many files in that folder you also have to find the image file by name manually. Reveal File in Folder may be a better command to implement - that's another issue but it would make sense when people really want to use that specific file to keep its original name - while Copy Image is just good enough to quickly share unnamed screenshots, etc.

**Comment by Krezzin** [comment]
It'd be a real helper if we can embed images along with text, instead of manually embedding the image through right-click > "Copy image". Maybe even have option in the settings or dunno, per-notebook option would work too.

**Comment by LbqpdL** [comment]
I also support this proposition.
